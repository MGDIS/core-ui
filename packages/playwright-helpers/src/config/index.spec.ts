import { beforeEach, describe, expect, test, vi } from 'vitest';

describe('playwrightBaseConfig', () => {
  beforeEach(() => {
    vi.resetModules();
  });

  test.each([false, true])('process.env.CI %s', async CI => {
    if (CI) import.meta.env.CI = CI;
    const playwrightBaseConfig = await import('.');
    expect(playwrightBaseConfig).toMatchSnapshot();
  });
});
