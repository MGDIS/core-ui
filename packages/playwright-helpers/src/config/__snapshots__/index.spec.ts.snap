// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`playwrightBaseConfig > process.env.CI false 1`] = `
{
  "default": {
    "expect": {
      "timeout": 5000,
    },
    "forbidOnly": false,
    "fullyParallel": true,
    "projects": [
      {
        "name": "chromium",
        "use": {
          "defaultBrowserType": "chromium",
          "deviceScaleFactor": 1,
          "hasTouch": false,
          "isMobile": false,
          "screen": {
            "height": 1080,
            "width": 1920,
          },
          "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.6099.28 Safari/537.36",
          "viewport": {
            "height": 720,
            "width": 1280,
          },
        },
      },
      {
        "name": "firefox",
        "use": {
          "defaultBrowserType": "firefox",
          "deviceScaleFactor": 1,
          "hasTouch": false,
          "isMobile": false,
          "screen": {
            "height": 1080,
            "width": 1920,
          },
          "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:119.0) Gecko/20100101 Firefox/119.0",
          "viewport": {
            "height": 720,
            "width": 1280,
          },
        },
      },
      {
        "name": "webkit",
        "use": {
          "defaultBrowserType": "webkit",
          "deviceScaleFactor": 2,
          "hasTouch": false,
          "isMobile": false,
          "screen": {
            "height": 1120,
            "width": 1792,
          },
          "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.4 Safari/605.1.15",
          "viewport": {
            "height": 720,
            "width": 1280,
          },
        },
      },
    ],
    "reporter": [
      [
        "html",
        {
          "open": "never",
        },
      ],
      [
        "junit",
        {
          "outputFile": "results.xml",
        },
      ],
    ],
    "retries": 0,
    "snapshotPathTemplate": "{testFileDir}/__screenshots__/{arg}-{projectName}{ext}",
    "testMatch": "*.e2e.ts",
    "timeout": 30000,
    "use": {
      "actionTimeout": 0,
      "headless": true,
      "trace": "on-first-retry",
    },
    "workers": undefined,
    Symbol(defineConfigWasUsed): true,
  },
}
`;

exports[`playwrightBaseConfig > process.env.CI true 1`] = `
{
  "default": {
    "expect": {
      "timeout": 5000,
    },
    "forbidOnly": true,
    "fullyParallel": true,
    "projects": [
      {
        "name": "chromium",
        "use": {
          "defaultBrowserType": "chromium",
          "deviceScaleFactor": 1,
          "hasTouch": false,
          "isMobile": false,
          "screen": {
            "height": 1080,
            "width": 1920,
          },
          "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.6099.28 Safari/537.36",
          "viewport": {
            "height": 720,
            "width": 1280,
          },
        },
      },
      {
        "name": "firefox",
        "use": {
          "defaultBrowserType": "firefox",
          "deviceScaleFactor": 1,
          "hasTouch": false,
          "isMobile": false,
          "screen": {
            "height": 1080,
            "width": 1920,
          },
          "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:119.0) Gecko/20100101 Firefox/119.0",
          "viewport": {
            "height": 720,
            "width": 1280,
          },
        },
      },
      {
        "name": "webkit",
        "use": {
          "defaultBrowserType": "webkit",
          "deviceScaleFactor": 2,
          "hasTouch": false,
          "isMobile": false,
          "screen": {
            "height": 1120,
            "width": 1792,
          },
          "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.4 Safari/605.1.15",
          "viewport": {
            "height": 720,
            "width": 1280,
          },
        },
      },
    ],
    "reporter": [
      [
        "html",
        {
          "open": "never",
        },
      ],
      [
        "junit",
        {
          "outputFile": "results.xml",
        },
      ],
    ],
    "retries": 2,
    "snapshotPathTemplate": "{testFileDir}/__screenshots__/{arg}-{projectName}{ext}",
    "testMatch": "*.e2e.ts",
    "timeout": 30000,
    "use": {
      "actionTimeout": 0,
      "headless": true,
      "trace": "on-first-retry",
    },
    "workers": 1,
    Symbol(defineConfigWasUsed): true,
  },
}
`;
